base-notebook/start.sh

```diff
@@ -40,7 +40,7 @@ if [ $(id -u) == 0 ] ; then
     # Ex: default NFS/EFS (no auto-uid/gid)
     if [[ "$CHOWN_HOME" == "1" || "$CHOWN_HOME" == 'yes' ]]; then
         echo "Changing ownership of /home/$NB_USER to $NB_UID:$NB_GID"
-        chown -R $NB_UID:$NB_GID /home/$NB_USER
+        chown $NB_UID:$NB_GID /home/$NB_USER
```

Comment:
The env var `CHOWN_HOME` setting is designed to me "chown the $NB_USER home directory recursively." If you think there's a need for a flag to indicate a non-recursive chown should be performed instead, I think we should be adding a env var instead of changing the meaning of the existing one.

Reply:
Please take a look at the git history.  I originally wrote this code due to work discussed in the zero to jupyterhub project.

The discussion about the chown can be found here: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/421#issuecomment-362019068

Doing a chown -R in a user's home directory is a terrible idea by default and will cause significant slowdowns as soon as the environment is seriously used.  We have <10 users and are already having issues due to this change.

If someone really want's to do a chown -R, a flag should be added for that.  However I think a better solution should be found, like specifying paths inside the user's home directory to be chowned.

Reply:
Sorry, clarifying: I thought the original intent of `CHOWN_HOME` was meant to be recursive and assumed (clearly, incorrectly) that the later PR from #576 was meant to correct the missing `-R` based on the conversation in the original issue #541.

I agree we either need two flags (`CHOWN_HOME`, `CHOWN_HOME_RECURSIVE`) or something more elegant.

Reply:
Breaking out the two flags would work, but I hope we can find something more elegant 

I wonder what @ewjoachim thinks about adding something like $CHOWN_HOME_SUBDIRS and have that be a list.  He has had the clearest use case for that flag so far.

On the other hand more options for the user is usually a good thing so we could do all 3 options, though I still worry that CHOWN_HOME_RECURSIVE can easily get users into trouble

Sorry if I took the offensive a bit too hard in my initial reply

Reply:
No worries. My initial response this morning was a too terse from tapping it out on a phone and lead to the initial confusion anyway.

`CHOWN_HOME_SUBDIRS` sounds good to me. And maybe if we support `.` as a value, it can mean the same thing as `CHOWN_HOME_RECURSIVE` without that additional option.

Reply:
I'm clearly 👍 on anything that fixes a problem I don't have while not creating problems for me 😄or if it breaks but there's a better workaround, I'll gladly take it too ! I've only made one contribution to the repo, so I'm far from being an expert on anything.

About the `CHOWN_HOME` being recursive or not, I just assumed it was supposed to be like this. If there's a better way to do what I did in #576 for #575, I'm totally ok to change my workflow ! Just let me know here or in the original ticket.

Reply:
@ewjoachim @parente I've pushed a commit to add CHOWN_EXTRA.  It should allow you to chown additional directories and get rid of the need to do a recursive chown